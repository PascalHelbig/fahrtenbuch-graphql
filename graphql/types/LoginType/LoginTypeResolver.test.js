const { makeExecutableSchema } = require('graphql-tools');
const { graphql } = require('graphql');
const LoginType = require('./LoginType');
const LoginTypeResolver = require('./LoginTypeResolver');

const SchemaDefinition = `
  type Query {
    login: Login
  }
      
  schema { query: Query }
`;

const testPropterty = (propterty) => {
  const resolvers = {
    Query: {
      login: () => ({
        user: { email: 'test@example.com' },
        token: 'the token',
      }),
    },
    Login: LoginTypeResolver,
  };
  const schema = makeExecutableSchema({
    typeDefs: [SchemaDefinition, LoginType],
    resolvers,
  });

  const query = `
    query { 
      login {
        ${propterty}
     }
    }
  `;

  return graphql(schema, query).then(result =>
    expect(result).toMatchSnapshot()
  );
};

it('should get the token from model', () =>
  testPropterty('token')
);

it('should get the user from model', () =>
  testPropterty('user { email }')
);
