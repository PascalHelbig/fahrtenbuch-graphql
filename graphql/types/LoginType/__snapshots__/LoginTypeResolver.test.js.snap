exports[`test should get the token from model 1`] = `
Object {
  "data": Object {
    "login": Object {
      "token": "the token",
    },
  },
}
`;

exports[`test should get the user from model 1`] = `
Object {
  "data": Object {
    "login": Object {
      "user": Object {
        "email": "test@example.com",
      },
    },
  },
}
`;
